libs/deer/src/impls/core/bytes.rs

```diff
@@ -0,0 +1,39 @@
+use core::marker::PhantomData;
+
+use error_stack::ResultExt;
+
+use crate::{
+    error::{DeserializeError, VisitorError},
+    Deserialize, Deserializer, Document, Reflection, Schema, Visitor,
+};
+
+struct BytesVisitor<'de>(PhantomData<fn() -> &'de ()>);
+
+impl<'de> Visitor<'de> for BytesVisitor<'de> {
+    type Value = &'de [u8];
+
+    fn expecting(&self) -> Document {
+        Self::Value::reflection()
+    }
+
+    fn visit_borrowed_bytes(self, v: &'de [u8]) -> error_stack::Result<Self::Value, VisitorError> {
+        Ok(v)
+    }
+}
+
+impl Reflection for [u8] {
+    fn schema(_: &mut Document) -> Schema {
+        // this type does not really exist in json-schema :/
+        // TODO: correct valid schema?
+        Schema::new("bytes")
```

Comment:
Do you have any ideas for how we could handle this in JSON Schema? And would it be problematic for us to be using `&[u8]` for JSON?

Reply:
`&[u8]` is invalid JSON; depending on the library, it is either rejected (JustJson) or strings that are not valid utf8 are passed as bytes (`serde-json`).

In the future, I plan to introduce a more robust system for schemas (project blueprint™️) that would solve this ambiguity by being agnostic over the output schema and having JSON-Schema be one of the output schemas.

For JSON-schema, specifically bytes data is typically encoded via `contentEncoding` (https://json-schema.org/understanding-json-schema/reference/non_json_data.html) so you would specify as encoding base64 and then deserialize into bytes.